File: backend/GoIsland.Api/Models/Review.cs
namespace GoIsland.Api.Models;

public static class ReviewModerationStatuses
{
    public const string Visible = "Visible";
    public const string Hidden = "Hidden";
    public const string Deleted = "Deleted";
    public const string Reported = "Reported";
    public static readonly string[] All = [Visible, Hidden, Deleted, Reported];
}

public class Review
{
    public int Id { get; set; }
    public int ReservationId { get; set; }
    public int UserId { get; set; }
    public int ExperienceId { get; set; }
    public int HostId { get; set; }
    public int Rating { get; set; }
    public string Comment { get; set; } = string.Empty;
    public string ModerationStatus { get; set; } = ReviewModerationStatuses.Visible;
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;
}

File: frontend/src/types/review.ts
export interface Review {
  id: number;
  reservationId: number;
  userId: number;
  authorName: string;
  experienceId: number;
  hostId: number;
  rating: number;
  comment: string;
  moderationStatus: string;
  createdAt: string;
  updatedAt: string;
}

export interface ReviewInput {
  rating: number;
  comment: string;
}

export type ReviewModerationStatus = 'Visible' | 'Hidden' | 'Deleted' | 'Reported';

export interface AdminReviewListParams {
  query?: string;
  status?: ReviewModerationStatus;
  page?: number;
  pageSize?: number;
}
